### PR TITLE
For #10502 - Show snackbar after adding tabs to collection from tab tray

### PR DESCRIPTION
**Pull request by darkwing**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayFragment.kt [hunk 1]
```diff
@@ -192,6 +205,9 @@ class TabTrayFragment : Fragment(R.layout.fragment_tab_tray), TabsTray.Observer,
 
         if (navController.currentDestination?.id == R.id.collectionCreationFragment) return
 
+        // Only register the observer right before moving to collection creation
+        registerCollectionStorageObserver()
```
Comment (darkwing) [inline app/src/main/java/org/mozilla/fenix/tabtray/TabTrayFragment.kt:209]: In HomeFragment we have an `unregister` in `onStart`-- should we be doing that here?  Adding an `unregister` in on `onStart` feels really weird.

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayFragment.kt [hunk 2]
```diff
@@ -201,6 +217,29 @@ class TabTrayFragment : Fragment(R.layout.fragment_tab_tray), TabsTray.Observer,
         navController.nav(R.id.tabTrayFragment, directions)
     }
 
+    private fun registerCollectionStorageObserver() {
+        requireComponents.core.tabCollectionStorage.register(collectionStorageObserver, this)
+    }
+
+    private fun showCollectionSnackbar() {
+        if (view != null) {
```
Comment (darkwing) [inline app/src/main/java/org/mozilla/fenix/tabtray/TabTrayFragment.kt:225]: Is there a way we can better solidify view so we aren't checking for it everywhere?

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayFragment.kt (hunk 1 above)
Comment (boek) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayFragment.kt:209]: I agree. I'm not sure if that actually helps :/

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayFragment.kt (hunk 2 above)
Comment (boek) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayFragment.kt:225]: I've been doing this
```
view?.also {
    
}
```

**Comment by darkwing** [comment]
Note to self: we may not need this if we move to the drawer.

**Pull request closed without merging**